tools/c7n_azure/c7n_azure/resources/network_interface.py

```diff
@@ -29,3 +33,25 @@ class resource_type(object):
             'location',
             'resourceGroup'
         )
+
+
+@NetworkInterface.filter_registry.register('effective-route-table')
+class EffectiveRouteTableFilter(ValueFilter):

```

Comment:
You can speed this up by following the model for threading it.  You implement the `process` method instead, you can reference https://github.com/cloud-custodian/cloud-custodian/blob/dcfa6cc6c7e1b7d15fdf21ef8949855665f1f4fa/tools/c7n_azure/c7n_azure/filters.py#L265